awscli/clidocs.py

```diff
@@ -186,18 +190,47 @@ def doc_relateditem(self, help_command, related_item, **kwargs):
         )
         doc.write('\n')
 
-    def _document_enums(self, argument, doc):
+    def _document_enums(self, model, doc):
         """Documents top-level parameter enums"""
-        if hasattr(argument, 'argument_model'):
-            model = argument.argument_model
-            if isinstance(model, StringShape):
-                if model.enum:
-                    doc.style.new_paragraph()
-                    doc.write('Possible values:')
-                    doc.style.start_ul()
-                    for enum in model.enum:
-                        doc.style.li('``%s``' % enum)
-                    doc.style.end_ul()
+        if isinstance(model, StringShape):
+            if model.enum:
+                doc.style.new_paragraph()
+                doc.write('Possible values:')
+                doc.style.start_ul()
+                for enum in model.enum:
+                    doc.style.li('``%s``' % enum)
+                doc.style.end_ul()
+
+    def _document_nested_structure(self, model, doc, stack):
+        """Recursively documents parameters in nested structures"""
+        shape_name = getattr(model, 'name', None)
+        if not shape_name or shape_name in stack:
+            return
+        stack.append(shape_name)
+        if isinstance(model, StructureShape):
+            self._output_struct_data(model.members, doc, stack)
+        if isinstance(model, ListShape):
+            self._output_list_data(model, doc, stack)
+        stack.pop()
```

Comment:
We should also account for map shapes. Here is how we do so in the output side: https://github.com/aws/aws-cli/blob/6a9c20c7b94f4bcab0ac577f7babfc35dd34cffc/awscli/clidocs.py#L561-L567